lib/engine/game/g_1849/step/buy_e_token.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+
+module Engine
+  module Game
+    module G1849
+      module Step
+        class BuyEToken < Engine::Step::Base
+          def actions(entity)
+            actions = super.dup
+            actions += %w[choose pass] if can_buy_e_token?(entity)
+
+            actions.uniq
+          end
+
+          def description
+            'Buy E-Token'
+          end
+
+          def pass_description
+            'Pass'
+          end
+
+          def choices
+            choices = []
+            choices << ["Buy E-Token for #{@game.format_currency(e_token_cost)}"] if can_buy_e_token?(current_entity)
+            choices
+          end
+
+          def choice_name
+            'E-Token allows the purchase of E-Trains'
+          end
+
+          def log_skip(entity)
+            super if @game.electric_dreams? && @game.phase.status.include?('gray_uses_black')
+          end
+
+          def auto_actions(entity)
+            return [Engine::Action::Pass.new(entity)] unless can_buy_e_token?(entity)
+
+            super
+          end
+
+          def e_token_cost
+            if @game.e_token_sold == true
+              @game.phase.name == '12H' ? 800 : 550
+            else
+              @game.phase.name == '12H' ? 1100 : 800
+            end
+          end
+
+          def process_choose(action)
+            raise GameError, 'Cannot buy E-token' unless can_buy_e_token?(action.entity)
+
+            buy_e_token(action.entity) if action.choice == "Buy E-Token for #{@game.format_currency(e_token_cost)}"
```

Comment:
There is only one choice, so you don't have to compare the output. And if you wanted to compare the output, you should generate that output in one place and use that in both `def choices` and here.